compiler/rustc_middle/src/ty/query.rs

```diff
@@ -230,21 +230,18 @@ macro_rules! define_callbacks {
                     Err(()) => (),
                 }
 
-                // self.tcx.queries.$name(self.tcx, DUMMY_SP, key, QueryMode::Ensure).unwrap()
+                // self.tcx.queries.$name(self.tcx, DUMMY_SP, key, QueryMode::Ensure);
                 let key = &key as *const query_keys::$name<'tcx>;
                 let mut out = MaybeUninit::<query_stored::$name<'tcx>>::uninit();
                 unsafe {
-                    let success = self.tcx.queries.execute(
+                    self.tcx.queries.execute(
                         self.tcx,
                         DUMMY_SP,
                         $query_id,
                         key.cast::<()>(),
                         out.as_mut_ptr().cast::<()>(),
```

Comment:
Would it be faster to skip writing the out value for TyCtxtEnsure?

Reply:
Hmm, maybe passing in a null pointer (and checking for that) would make sense, I can try that